language/move-ir/types/src/location.rs

```diff
@@ -16,46 +16,53 @@ use std::{
 #[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize, Hash)]
 pub struct Loc {
     file: &'static str,
-    span: Span,
+    start: u32,
+    end: u32,
```

Comment:
Out of curiosity, did you consider implementing these two fields as a single [`std::ops::Range<u32>`](https://doc.rust-lang.org/std/ops/struct.Range.html) (or `Range<ByteIndex>`)? Especially since you include that type in the file above anyway, it seems like a natural fit, and it includes well-documented contracts on whether the range is open, what constitutes an empty range, etc. Did you opt against it for performance or size concerns?

Also, I assume a `u32` here for the same reasons as codespan, size efficiency and an assumption source files won't exceed 4GB? https://docs.rs/codespan/0.11.1/src/codespan/index.rs.html#8-10 If so that makes sense to me, but I think a comment documenting the size limit would be good to add. (Also maybe some of the spots casting between `usize` and `u32` could be cleaned up...)

Reply:
I tried, but `Range<u32>` oddly doesn't implement `Copy` 

And will definitely add a comment about file sizes